en/guide/extending_csd.md

```diff
@@ -1,6 +1,121 @@
 # Extending Camera Streaming Daemon
 
-## Create a Custom Video Stream
+## Create a Custom Camera Device
+
+Camera-Streaming-Daemon has native support for auto detection of video cameras exported as Video4Linux(V4L2) devices. This covers most of regular Linux cameras, but there are some cameras that don't have V4L2 support or cases where user wants to set very specific parameters or do custom post-processing in video before exporting.
+
+For these use cases, it is possible to create a custom CameraDevice class, representing a single custom camera device.
+CameraServer class keeps a list of all created CameraComponents, each one holding a single available CameraDevice. These CameraDevice objects are used by CameraComponent class for camera control and ImageCapture and VideoStream classes to perform the image capture and streaming.
+
+[CameraDeviceGazebo](https://github.com/intel/camera-streaming-daemon/blob/master/src/CameraDeviceGazebo.cpp) is an example of a custom CameraDevice. We will explore it in order to learn all the steps necessary to add new custom CameraDevice in Camera-Streaming-Daemon.
+
+### CameraDevice
+
+CameraDevice is the abstract base class for all camera devices.
+
+```cpp
+class CameraDevice {
+```
+Every CameraDevice is identified by a unique string. For V4L2 class of camera devices, its the device node videox (For example video0 ). For Gazebo, its the string "gazebo".
```

Comment:
Having an empty class outline like above isn't useful/doesn't tell us anything new. If this is to be kept it should show something we can't guess from the name - maybe an example of the string definition, or the methods that must be overloaded.

> For V4L2 class of camera devices, its the device node videox (For example video0 )

This seems to imply that there are multiple different V4L2 devices, and we'd give them names like video0, video1 etc. If so, what would be the difference between these types of devices, and what types do we actually support currently. It is possible I am misinterpreting this, and each `CameraDevice` *instance* gets its own string (?)

Reply:
8. Shall replace empty class outline with more details

> This seems to imply that there are multiple different V4L2 devices, and we'd give them names like video0, video1 etc. If so, what would be the difference between these types of devices, and what types do we actually support currently. It is possible I am misinterpreting this, and each CameraDevice instance gets its own string (?)

There can be multiple v4l2 devices and an instance of  `CameraComponent` (and `CameraDeviceV4l2` ) for each of them identified by string like video0, video1.  It is possible that there is a v4l2 device that we do not support, need to blacklist that in conf file. We support v4l2 standard compliant streaming devices.

Reply:
OK, so do we have a list of the videox classes we already support - we should probably list those that we support so that people don't try to do their own implementations (does this make sense?)

Reply:
16. Shall add list of v4l2 standard compliant cameras